turbomoleio/testfiles/utils.py

```diff
@@ -298,6 +298,136 @@ def assert_almost_equal(actual, desired, rtol=1e-7, atol=0, ignored_values=None,
             raise
 
 
+REF_DICT_TEST_OTHER = 'REF_DICT_TEST_OTHER'
+REF_SEQUENCE_TEST_OTHER = 'REF_SEQUENCE_TEST_OTHER'
+REF_STRING_TEST_OTHER = 'REF_STRING_TEST_OTHER'
+TEST_STRING_REF_OTHER = 'TEST_STRING_REF_OTHER'
+DICT_DIFFERENT_KEYS = 'DICT_DIFFERENT_KEYS'
+SEQUENCE_DIFFERENT_SIZES = 'SEQUENCE_DIFFERENT_SIZES'
+NUMBERS_DIFFER = 'NUMBERS_DIFFER'
+STRINGS_DIFFER = 'STRINGS_DIFFER'
+DIFFERENCE_TYPES = [REF_DICT_TEST_OTHER, REF_SEQUENCE_TEST_OTHER,
+                    REF_STRING_TEST_OTHER, TEST_STRING_REF_OTHER, DICT_DIFFERENT_KEYS,
+                    SEQUENCE_DIFFERENT_SIZES, NUMBERS_DIFFER, STRINGS_DIFFER]
+
+
+def _update_differences(differences, level, difference_type, message=None):
+    if difference_type not in DIFFERENCE_TYPES:
+        raise ValueError(f'Difference type "{difference_type}" is not valid')
+    msg = f'>>>{difference_type}<<<'
+    if message is not None:
+        msg += f'\n{message}'
+    level_t = tuple(level)
+    if level_t in differences:
+        raise RuntimeError('Should not be reached here')
+    differences[level_t] = msg
+
+
+def compare_differences(actual, desired, rtol=1e-7, atol=0, current_level=None):
```

Comment:
Should this include an `ignored_values` argument like `assert_almost_equal`. Could it also be an argument for ITestConfig? If during a test one gets a huge list of differences, could it be good to ignore some of them temporarily?

Reply:
Maybe I would add that later if it is needed. What do you think ?

Reply:
it can be ok to postpone if you already tested this on some TM versions and you didn't feel the need for it

Reply:
ok, I'll postpone at this stage.